### ISSUE TITLE
Improvements to the Read Timer

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#9216]
**CONTEXT:**
Read timers are important. Because looking at screens for a long time causes eye strain and eyes need rest at regular intervals. American Academy of Ophthalmology suggests following the “20-20-20” rule which is taking regular breaks every 20 minutes, shifting your eyes to look at an object at least 20 feet away, for at least 20 seconds.

This is so important we have tons of screen timer/work timer apps on every platform. It is also integrated to the reading apps naturally. And as a computer user you probably know this by personal experience (fatigued, dry eyes etc.).

**PROBLEM:**
KOReader has a read timer too. But our one has a problem. 
It is an egg timer. Not a wake up timer. Let me explain.

There are two types of alarms:
- **Recurring alarms:** If you wake up every day at 8 for work, this is a recurring alarm. You don't want to set it every day for the next day. You set it once and expect it to ring until you disable it (during your next vacation for example).
- **Single use alarms:** Egg or tea timers are like this. You set it once for a task. Once your egg is cooked you are done. You don't want it to ring every 6 minutes after your egg is cooked.

As you noticed by now, a read timer is a recurring alarm. Because our eyes need rest every 20 minutes of screen work for the foreseeable future (probably until we die).

**A proper implementation example:**
In Moon Reader (my other favorite reader app) reading timer procedure is just one step:
- Go to settings and enable Read Timer with a checkbox (optionally change the interval)

Now whenever you open a book, it starts the timer and every X minutes it shows you a message for 60 seconds.
You set it once and it works continuously which is what you expect from a recurring timer. Set and forget.

**How a proper read timer should be implemented:**
- In the settings there is an option to enable the timer globally which survives restarts
- Whenever user opens a book, 
1.timer starts automatically  2.shows a popup at the set interval 3.then start again
- Settings can be simply like this: "Remind me every __ minutes for a duration of __ seconds (optional>) with this message _____"
- If user suspends the device, **reset** the timer because this counts as a **rest**. User looked away from the device.
- If user opens another document **continue counting** (don't interrupt or reset) because user **didn't** take rest, he is still looking at the screen.

**For the reminder a popup like this:**
![readtimerdialog](https://user-images.githubusercontent.com/84420582/173811105-92a6ac4b-2314-4f0d-8fdc-73d27aff8ea6.png)

**SKIP THIS REST:** In the case user is not tired yet or just want to continue reading
**DISABLE TIMER:** In the case user is in a hurry and don't want to be interrupted, for example while preparing for an exam so he can disable it quickly without going into menu

**Optionally:** Screen can be inverted during break so user can check from far away (looking out the window maybe) if break time is over without coming near the device.

**Fancy upgrade:** Short/long rest intervals. Every 20 minutes for 20 sec. Every hour for 5 minutes.

Of course details are open to discussion but I think you get the general idea and reasoning behind my suggestions. 
Thanks for coming to my TED talk.

---

Also I can't finish without mentioning that this is one of the funniest dialogs I have ever seen. A read timer I set 20 minutes ago reminding me of full date including which year I am in. "Hey it's 2022, take a rest buddy !" :)))

![readtimerpopup](https://user-images.githubusercontent.com/84420582/173814853-21de7593-e4ff-4dfa-87db-0b5b514e2a53.png)

**Comment by roygbyte** [comment]
A different UI approach could show the timer's progress in the status bar, much like how chapter progress is shown. I suggest this for people like me who like to anticipate when an interruption will arrive/a task will end, instead of having the interruption appear without warning.

**Comment by offset-torque** [comment]
Yeah that would be nice too, especially for people using their devices for studying. With a countdown timer they can do pomodoro like study intervals directly on the device.

**Comment by offset-torque** [comment]
> But then my use is something closer to "it's midnight and I want to be reminded to see if I want to sleep in 15 minutes" or "I'm on the train and in 10 minutes I want to start paying attention to stations etc."

Ok I see, more like the **Sleep Timer** feature in audio player apps.
If that's the intended use case, functionality is enough I have to agree.
But naming can be disappointing for people coming from other e-reader apps expecting the same type of functionality (a recurring eye rest reminder).

**Comment by greatyingzi** [comment]
It will support repeat, when interval end. https://github.com/koreader/koreader/pull/9844

**Comment by offset-torque** [comment]
### Status update
I see that changes are made to the read timer to address the problems in this issue:

- [x] We now have a dialog (as I suggested), which allows repeating/ending the timer without going into the menu

- [x] Also now remanining time can be seen in the status bar as **roygbyte** suggested which I think is useful

Thanks for these improvements. Our read timer is not an "egg timer" anymore :)

---

**However my main improvement item still remains, regarding the activation of the timer:**
It still takes 6 taps **per book** to activate the read timer. Which means, it has to be activated manually when the user opens a book and this has to be repeated if user opens another book.

It would be nicer if it can be globally set, as I suggested in the main text of this issue and summarized here:
- There should be an option to enable the timer globally which survives restarts and switching books
- If user opens another document, continue counting
- If user suspends the device, reset the timer

**Comment by bejp0a** [comment]
Has anyone been able to figure out how to use either a profile or the http events server to auto-start the read timer?

**Comment by jwillikers** [comment]
> Has anyone been able to figure out how to use either a profile or the http events server to auto-start the read timer?

@bejp0a I forked the plugin and configured [my own version](https://github.com/jwillikers/koreader-break-timer) that runs automatically. It's not perfect, but it forces me to take breaks by putting a dialogue over the screan.

**Comment by shuvashish76** [comment]
> Yeah that would be nice too, especially for people using their devices for studying. With a countdown timer they can do pomodoro like study intervals directly on the device.

My use-case is mostly about Pomodoro. I'm using [this](https://f-droid.org/en/packages/com.apps.adrcotfas.goodtime/) android app for various purposes. Inbuilt pomodoro like feature would be nice
It which has two modes

1. Countdown mode
  - Focus time: Usually around 25mins for each session
  - [x] Break time (optional): ~5min
  - [ ] Long break (optional): ~15min
    - Sessions before long break : 4
2. Count up mode : Described [here](https://www.lesswrong.com/posts/RWu8eZqbwgB9zaerh/third-time-a-better-way-to-work)
